odd-platform-api/src/main/java/com/provectus/oddplatform/service/AlertLocatorImpl.java

```diff
@@ -0,0 +1,173 @@
+package com.provectus.oddplatform.service;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.provectus.oddplatform.api.contract.model.AlertStatus;
+import com.provectus.oddplatform.api.contract.model.AlertType;
+import com.provectus.oddplatform.dto.DataConsumerAttributes;
+import com.provectus.oddplatform.dto.DataEntitySpecificAttributesDelta;
+import com.provectus.oddplatform.dto.DataEntityType;
+import com.provectus.oddplatform.dto.DataTransformerAttributes;
+import com.provectus.oddplatform.dto.DatasetStructureDelta;
+import com.provectus.oddplatform.dto.IngestionTaskRun;
+import com.provectus.oddplatform.model.tables.pojos.AlertPojo;
+import com.provectus.oddplatform.model.tables.pojos.DatasetFieldPojo;
+import com.provectus.oddplatform.utils.JSONSerDeUtils;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.collections4.SetUtils;
+import org.springframework.stereotype.Component;
+
+import static java.util.function.Function.identity;
+
+@Component
+@RequiredArgsConstructor
+public class AlertLocatorImpl implements AlertLocator {
+    private static final Set<IngestionTaskRun.IngestionTaskRunStatus> TASK_RUN_BAD_STATUSES = Set.of(
+        IngestionTaskRun.IngestionTaskRunStatus.BROKEN,
+        IngestionTaskRun.IngestionTaskRunStatus.FAILED
+    );
+
+    @Override
+    public List<AlertPojo> locateDatasetBIS(final Map<String, DatasetStructureDelta> structureDeltas) {
+        return structureDeltas.entrySet()
+            .stream()
+            .flatMap(this::locateAlertsInDSDelta)
+            .collect(Collectors.toList());
+    }
+
+    @Override
+    public List<AlertPojo> locateDQF(final List<IngestionTaskRun> taskRuns) {
```

Comment:
What is F in "DFQ"? Sometimes where are too many unknown abbreviations. It might be a good idea to add some basic javadoc for the methods with such abbreviations.

Reply:
Expanded a bit.

BIS: Backward Incompatible Schema
DQF: DataQualityTest Failed